### DISCUSSION TITLE
Does pocketbase support Shared Storage for multiple instances? (k8s pods)

### DISCUSSION THREAD

**Discussion by nisimasacode** [pocketbase/pocketbase#5601] [Q&A]
Dear community,

I mange the backend for a simple web application that relies on pocketbase. I sucessfully deployed the app on a Linode kubernetes cluster, but am wondering about how to mange the volumes: Does pocketbase support Shared Storage for multiple instances? When I have for example 3 instances of pocketbase running can they all share the same volume or will this lead to problems?

So basically the question is, can something like [this](https://kubernetes.io/docs/tasks/run-application/run-replicated-stateful-application/) be done with pocketbase? 

Thank you!

**Comment by whmeitzler** [comment]
Pocketbase is designed to be deployed as a single binary on a single server.  It doesn't make sense to force Pocketbase into a situation where it is expected to scale horizontally.  That said, it is also extremely performant, and vertical scaling is a helluva lot simpler to manage. Maybe consider if you need the complexity of K8s and hortizantal scaling in the first place?

**Comment by maietta** [comment]
Hi @whmeitzler.

I'm looking at SQLite's new proposal for native replication features. This, paired with SQLite's new proposed data modification hooks API could become the basis for a future where you can deploy more instances and keep databases in sync.

**Comment by whmeitzler** [comment]
Hey Nick,

That’s great (and it’s cool to see SQLite exploring distributed use cases),
but that doesn’t change the fact that pocketbase was designed for a
lockstep 1:1 relationship between service and DB.

Hooks, custom routes, and realtime are all evaluated and managed within
pocketbase. There is no mechanism to sync these actions across multiple
instances. I can only speculate that auth would be ok as it uses JWT,  but
that’s something you’d really want to vet.

If you’re not using realtime or hooks, then you can probably do something,
but at that point, you’re fighting Pocketbases core design and may benefit
from looking at something like Supabase.

Or you can ride the wave and see how far vertical scaling takes you. Your
call!


-Will

**Reply by nisimasacode** [reply]
Hi Will.

Thank you for your insights! I will resort to scaling pocketbase vertically if needed and in case we will actually need a horizontally scalable database I will work with the developer on migrating to another option.
